clang/test/Sema/nested-redef.c

```diff
@@ -19,4 +19,8 @@ void f2(void) {
   struct U u;
 }
 
-
+void f3(void) {
```

Comment:
So unlike the original test case this one does not seem to cause a crash. Can we add a test case for that as well to verify we are fixing that issue as well.